test/integration/evictions/evictions_test.go

```diff
@@ -0,0 +1,242 @@
+				Phase: v1.PodRunning,
+				Conditions: []v1.PodCondition{
+					{
+						Type:   v1.PodReady,
+						Status: v1.ConditionTrue,
+					},
+				},
+			},
+		}
+
+		if _, err := clientSet.Core().Pods(ns.Name).Create(pod); err != nil {
+			t.Errorf("Failed to create pod: %v", err)
+		}
+
+		clientSet.Core().Pods(ns.Name).UpdateStatus(pod)
+	}
+
+	waitToObservePods(t, podInformer, 10)
+
+	pdb := &v1beta1.PodDisruptionBudget{
+		ObjectMeta: v1.ObjectMeta{
+			Name: fmt.Sprintf("test-pdb"),
+		},
+		Spec: v1beta1.PodDisruptionBudgetSpec{
+			MinAvailable: intstr.IntOrString{
+				Type:   intstr.Int,
+				IntVal: 0,
+			},
+			Selector: &unversioned.LabelSelector{
+				MatchLabels: map[string]string{"app": "test-evictions"},
+			},
+		},
+	}
+	if _, err := clientSet.Policy().PodDisruptionBudgets(ns.Name).Create(pdb); err != nil {
+		t.Errorf("Failed to create PodDisruptionBudget: %v", err)
+	}
+
+	waitPDBStable(t, clientSet, 10, ns.Name, pdb.Name)
+
+	doneCh := make(chan bool, 10)
+	errCh := make(chan error, 1)
+	// spawn 10 goroutine to concurrently evict the pods
+	for i := 0; i < 10; i++ {
+		go func(id int, doneCh chan bool, errCh chan error) {
+			eviction := &v1beta1.Eviction{
+				TypeMeta: unversioned.TypeMeta{
+					APIVersion: "Policy/v1beta1",
+					Kind:       "Eviction",
+				},
+				ObjectMeta: v1.ObjectMeta{
+					Name:      fmt.Sprintf(podNameFormat, id),
+					Namespace: ns.Name,
+				},
+				DeleteOptions: deleteOption,
+			}
+
+			var e error
+			for {
+				e = clientSet.Policy().Evictions(ns.Name).Evict(eviction)
+				if errors.IsTooManyRequests(e) {
+					time.Sleep(5 * time.Second)
+				} else {
+					break
+				}
+			}
+			if e != nil {
+				if errors.IsConflict(err) {
+					errCh <- fmt.Errorf("Failed to handle concurrent updates on PDB because of conflict: %v", e)
+				} else {
+					errCh <- e
+				}
+				return
+			}
+			doneCh <- true
+		}(i, doneCh, errCh)
+	}
+
+	doneCount := 0
+	for {
+		select {
+		case err := <-errCh:
+			t.Errorf("%v", err)
+			return
+		case <-doneCh:
+			doneCount++
+			if doneCount == 10 {
+				return
+			}
+		case <-time.After(defaultTimeout):
+			t.Errorf("Eviction did not complete within %v", defaultTimeout)
+		}
+	}
+
+	for i := 0; i < 10; i++ {
+		podName := fmt.Sprintf(podNameFormat, i)
+		_, err := clientSet.Core().Pods(ns.Name).Get(podName)
+		if !errors.IsNotFound(err) {
+			t.Errorf("Pod %q is expected to be evicted", podName)
+		}
+	}
+
+	if err := clientSet.Policy().PodDisruptionBudgets(ns.Name).Delete(pdb.Name, deleteOption); err != nil {
+		t.Errorf("Failed to delete PodDisruptionBudget: %v", err)
+	}
+
+	close(stopCh)
+}
+
+// wait for the podInformer to observe the pods. Call this function before
+// running the RS controller to prevent the rc manager from creating new pods
+// rather than adopting the existing ones.
+func waitToObservePods(t *testing.T, podInformer cache.SharedIndexInformer, podNum int) {
+	if err := wait.Poll(10*time.Second, 60*time.Second, func() (bool, error) {
```

Comment:
PollImmediate. Having to wait for 10s before doing the first check is a long time.